src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -95,6 +96,120 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
+            {
+                return SubstiteParameterSymbols(mergedReturnType, mergedParameterTypes);
+            }
+            else
+            {
+                return this;
+            }
+        }
+
+        public FunctionPointerMethodSymbol SetNullabilityForReferenceTypes(Func<TypeWithAnnotations, TypeWithAnnotations> transform)
+        {
+            var transformedParameterTypes = ImmutableArray<TypeWithAnnotations>.Empty;
+            bool hasParamChanges = false;
+            if (_parameters.Length > 0)
+            {
+                var builder = ArrayBuilder<TypeWithAnnotations>.GetInstance(_parameters.Length);
+                foreach (var param in _parameters)
+                {
+                    var transformedType = transform(param.TypeWithAnnotations);
+                    builder.Add(transformedType);
+                    if (!transformedType.IsSameAs(param.TypeWithAnnotations))
+                    {
+                        hasParamChanges = true;
+                    }
+                }
+
+                transformedParameterTypes = builder.ToImmutableAndFree();
+            }
+
+            var transformedReturn = transform(ReturnTypeWithAnnotations);
+            if (hasParamChanges || !transformedReturn.IsSameAs(ReturnTypeWithAnnotations))
+            {
+                return SubstiteParameterSymbols(transformedReturn, transformedParameterTypes);
+            }
+            else
+            {
+                return this;
+            }
+        }
+
+        private FunctionPointerMethodSymbol(
+            CallingConvention callingConvention,
+            RefKind refKind,
+            TypeWithAnnotations returnType,
+            ImmutableArray<CustomModifier> refCustomModifiers,
+            ImmutableArray<ParameterSymbol> originalParameters,
+            ImmutableArray<TypeWithAnnotations> substitutedParameterTypes)
+        {
+            Debug.Assert(originalParameters.Length == substitutedParameterTypes.Length);
+            RefCustomModifiers = refCustomModifiers;
+            CallingConvention = callingConvention;
+            RefKind = refKind;
+            ReturnTypeWithAnnotations = returnType;
+
+            if (originalParameters.Length > 0)
+            {
+                var paramsBuilder = ArrayBuilder<FunctionPointerParameterSymbol>.GetInstance(originalParameters.Length);
+                for (int i = 0; i < originalParameters.Length; i++)
+                {
+                    var originalParam = originalParameters[i];
+                    var substitutedType = substitutedParameterTypes[i];
+                    paramsBuilder.Add(new FunctionPointerParameterSymbol(
+                        substitutedType,
+                        originalParam.RefKind,
+                        originalParam.Ordinal,
+                        containingSymbol: this,
+                        originalParam.RefCustomModifiers));
```

Comment:
>originalParam.RefCustomModifiers [](start = 24, length = 32)

I think these are subject for substitution as well?

Reply:
Comment added to tracking issue.

---
In reply to: [406851037](https://github.com/dotnet/roslyn/pull/43041#discussion_r406851037) [](ancestors = 406851037)